packages/core/src/httpClient/retryableHttpClient.ts

```diff
@@ -31,4 +93,18 @@ export class RetryableHttpClient extends BasicHttpClient {
       retryCount++;
     }
   }
+
+  protected async request<ResponseType>(request: RetryableHttpRequest) {
+    return super.request<ResponseType>(request);
+  }
 }
+
+export type RetryableHttpRequest = HttpRequest &
+  HttpRequestRetryValidatorOptions;
+
+export type RetryableHttpRequestOptions = HttpRequestOptions &
+  HttpRequestRetryValidatorOptions;
+
+type HttpRequestRetryValidatorOptions = {
+  retryValidator?: false | RetryValidator;
```

Comment:
So the behaviour is as follows:
- `null`/`undefined` -> default retrier
- `false` -> no retries at all
- `function` -> custom retrier
?

Reply:
Yes, the motivation was flexibility - but of course that comes with a small cost on cognitive complexity.

We can even go one step further and add the option of numbers for retry counts in the future as well - sth like this: https://tanstack.com/query/v4/docs/guides/query-retries. I didn't want to have it as I was forcing a max retry count - but if we won't have it, we can have that option as well. In that case the behaviour would be:

- undefined-> default retry validator
- false -> no retry
- function -> custom retry validator
- number -> just check retry count and retry if it doesn't exceed the argument